src/cogent3/app/data_store_new.py

```diff
@@ -0,0 +1,340 @@
+import glob
+import re
+import shutil
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from functools import singledispatch
+from pathlib import Path
+
+from scitrack import get_text_hexdigest
+
+from cogent3.util.io import open_
+from cogent3.util.parallel import is_master_process
+
+
+__author__ = "Gavin Huttley"
+__copyright__ = "Copyright 2007-2022, The Cogent Project"
+__credits__ = ["Gavin Huttley"]
+__license__ = "BSD-3"
+__version__ = "2022.8.24a1"
+__maintainer__ = "Gavin Huttley"
+__email__ = "Gavin.Huttley@anu.edu.au"
+__status__ = "Alpha"
+
+_INCOMPLETE_TABLE = "incomplete"
```

Comment:
Thinking about this more I think `_NOT_COMPLETED_TABLE` is a better name. The phrase "not completed" indicates something could not be completed. Incomplete implies it's interrupted and might be yet completed. We want the former and not the latter. Refactor to change all occurrences throughout.